dds/DCPS/ReactorInterceptor.cpp

```diff
@@ -43,16 +44,22 @@ int ReactorInterceptor::handle_exception(ACE_HANDLE /*fd*/)
 
 void ReactorInterceptor::process_command_queue_i()
 {
-  ACE_GUARD(ACE_Thread_Mutex, guard, mutex_);
+  OPENDDS_QUEUE(CommandPtr) cq;
   ACE_Reverse_Lock<ACE_Thread_Mutex> rev_lock(mutex_);
+
+  ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
+  state_ = PROCESSING;
   while (!command_queue_.empty()) {
-    CommandPtr command = command_queue_.front();
-    command_queue_.pop();
-    ACE_GUARD(ACE_Reverse_Lock<ACE_Thread_Mutex>, rev_guard, rev_lock);
-    command->execute();
-    command->executed();
-    command.reset();
+    cq.swap(command_queue_);
+    ACE_Guard<ACE_Reverse_Lock<ACE_Thread_Mutex> > rev_guard(rev_lock);
+    while (!cq.empty()) {
+      CommandPtr command = cq.front();
+      cq.pop();
+      command->execute();
+      command->executed();
+    }
   }
+  state_ = NONE;
 }
```

Comment:
I believe this is fundamentally flawed.  At one point, there was logic that measure the size of the queue at the beginning and only executed that many commands.  This was to prevent potential live lock where a command repeatedly puts itself on the queue.  I would recommend doing the swap first, process the queue, and then check the queue again after the loop to see if it needs to notify itself again.

Reply:
That's a good consideration, but I think it's beyond the scope of this PR. You're arguing for a design change of the current ReactorInterceptor (to avoid starvation of the non-ReactorInterceptor reactor tasks), but the change in this PR intentionally preserves the current command execution behavior (run until the member queue is empty) while preventing unnecessary calls to notify by using a state flag and using queue::swap (which is evidently C++11, so I guess we switch to dequeue) in order to lock / contend the mutex less frequently.